lib/parser/objects.ts

```diff
@@ -901,49 +901,70 @@ function* iterateContexts(object: ObjectBase & I.IHasContextReference, directlyV
 }
 // Returns all object visible starting from the startObjects scope.
 // The second parameter defines if the object is directly visible.
-export function* scope(startObject: ObjectBase, elaborateNames: ElaborateNames|undefined = undefined): Generator<[ObjectBase, boolean]> {
+export function* scope(startObject: ObjectBase, elaborateNames: ElaborateNames | undefined = undefined): Generator<[ObjectBase, boolean]> {
   let current = startObject;
   let directlyVisible = true;
+  // make sure stuff is only yielded once
+  const alreadyYieldedDirectlyVisible: ObjectBase[] = [];
+  const alreadyYieldedNoneDirectlyVisible: ObjectBase[] = [];
+  function* yieldWrapper([obj, directlyVisible]: [ObjectBase, boolean]): Generator<[ObjectBase, boolean]> {
```

Comment:
I don't like the function declaration inside a function. Maybe the scope thingy can be extracted into a seperate class (maybe also in another file)